### DISCUSSION TITLE
Multi-step shows extra "undefined" steps for the first seconds

### DISCUSSION THREAD

**Discussion by notflip** [formkit/formkit#1040] [General]
See video, in which I refresh the page 3 times to show this issue: https://jmp.sh/ifJKvViM

**Comment by justin-schroeder** [comment]
This sounds like an issue with SSR on nuxt. How are you steps being defined? We try to "sniff" out the step names during SSR in Nuxt by pre-rendering the children and extracting their step names in the parent. If those names are not available when that pre-render is done you would get the exact symptom you are showing there.

**Comment by notflip** [comment]
@justin-schroeder Thanks for getting back, the multi-step names are just hard-coded

```
<FormKit type="step" name="step-1" label="Start" next-label="Start">
...
<FormKit type="step" name="step-2" label="Pijn" previous-label="Vorige stap" next-label="Volgende stap">
...
<FormKit type="step" name="step-3" label="Blessure" previous-label="Vorige stap" next-label="Volgende stap">
```

**Comment by notflip** [comment]
@justin-schroeder Can you think of any workaround I can use for now? Thanks

**Comment by notflip** [comment]
@justin-schroeder Can you think of any workaround I can use for now? Thanks

**Comment by justin-schroeder** [comment]
You could just wrap it in `<ClientOnly>` for now
